lib/PuppeteerSharp.Tests/PuppeteerLoaderFixture.cs

```diff
@@ -45,64 +25,27 @@ public void Dispose()
             var serverTask = StartWebServerAsync();
 
             await Task.WhenAll(downloaderTask, serverTask);
-
         }
 
         private async Task StartWebServerAsync()
         {
-            var taskWrapper = new TaskCompletionSource<bool>();
-            const int timeout = 2000;
-
-            var build = Directory.GetCurrentDirectory().Contains("Debug") ? "Debug" : "Release";
-            var webServerPath = Path.Combine(Directory.GetCurrentDirectory(), "..", "..", "..", "..",
-                                             "PuppeteerSharp.TestServer");
-
-            _webServerProcess = new Process();
-            _webServerProcess.StartInfo.UseShellExecute = false;
-            _webServerProcess.StartInfo.FileName = "dotnet";
-            _webServerProcess.StartInfo.WorkingDirectory = webServerPath;
-            _webServerProcess.StartInfo.Arguments = $"./bin/{build}/netcoreapp2.0/PuppeteerSharp.TestServer.dll";
+            var builder = TestServer.Program.GetWebHostBuilder();
 
-            _webServerProcess.StartInfo.RedirectStandardOutput = true;
-            _webServerProcess.StartInfo.RedirectStandardError = true;
+            builder.UseContentRoot(FindParentDirectory("PuppeteerSharp.TestServer"));
 
-            _webServerProcess.OutputDataReceived += (sender, e) =>
-            {
-                Console.WriteLine(e.Data);
-                if (e.Data != null &&
-                    taskWrapper.Task.Status != TaskStatus.RanToCompletion &&
-                    //Though this is not bulletproof for the purpose of local testing
-                    //We assume that if the address is already in use is because we have another
-                    //process hosting the site
-                    (e.Data.Contains("Now listening on") || e.Data.Contains("ADDRINUSE")))
-                {
-                    taskWrapper.SetResult(true);
-                }
-            };
+            _host = builder.Build();
 
-            _webServerProcess.Exited += (sender, e) =>
-            {
-                taskWrapper.SetException(new Exception("Unable to start web server"));
-            };
+            await _host.StartAsync();
+        }
 
-            Timer timer = null;
-            //We have to declare timer before initializing it because if we don't do this 
-            //we can't dispose it in the action created in the constructor
-            timer = new Timer((state) =>
+        public static string FindParentDirectory(string directory)
```

Comment:
My comment got lost due to the changes. I'd create a TestUtils.cs class with this. We might need it for other generic stuff.